### ISSUE TITLE
update for AssemblyScript 0.18.*?

### ISSUE THREAD

**Issue by bcolloran** [jtenner/as-pect#328]
the instructions for installing as-pect at https://tenner-joshua.gitbook.io/as-pect/getting-started (using assemblyscript nightly) result in:

```
$ npm install --save-dev @as-pect/cli
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: assemblyscript@0.10.0-nightly.20200611
npm ERR! node_modules/assemblyscript
npm ERR!   dev assemblyscript@"0.10.0-nightly.20200611" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! dev @as-pect/cli@"*" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: assemblyscript@0.17.14
npm ERR! node_modules/assemblyscript
npm ERR!   peer assemblyscript@"^0.17.10" from @as-pect/cli@5.0.1
npm ERR!   node_modules/@as-pect/cli
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Perhaps the version numbers in package.json need to be bumped? (If as-pect is compatible with the latest versions of course)

**Comment by bcolloran** [comment]
More information:

Unfortunately, it appears that there is genuine incompatibility between 0.17.* and 0.18.*, and more than a version bump will be required -- after installing as-pect according to the quick start instructions, but with `npm install --legacy-peer-deps --save-dev @as-pect/cli` and trying to the example tests in `example.spec.ts`, I'm encountering:

```
[Error] There was a compilation error when trying to create the wasm binary for file: assembly/__tests__/example.spec.ts.
Error: Runtime 'full' not found.
```

Looking at the docs about (runtimes)[https://www.assemblyscript.org/runtime.html] and the new [garbage collection implementation](https://www.assemblyscript.org/garbage-collection.html#runtime-interface) appears that the acceptable values for the `--runtime` flag in `as-pect.config.js` should now be "incremental", "minimal", and "stub".

However, it seems that there may be additional updates required, because when I try any of those values, I get the error:

```
$ npx  asp --verbose
       ___   _____                       __    
      /   | / ___/      ____  ___  _____/ /_   
     / /| | \__ \______/ __ \/ _ \/ ___/ __/   
    / ___ |___/ /_____/ /_/ /  __/ /__/ /_     
   /_/  |_/____/     / .___/\___/\___/\__/     
                    /_/                        

⚡AS-pect⚡ Test suite runner [5.0.1]

[Log] Loading asc compiler
Assemblyscript Folder:assemblyscript
[Log] Compiler loaded in 450.581ms.
[Log] Using configuration ~/asmscript_test/as-pect.config.js
[Log] Using VerboseReporter
[Log] Including files: assembly/__tests__/**/*.spec.ts
[Log] Running tests that match: (:?)
[Log] Running groups that match: (:?)
[Log] Effective command line args:
  [TestFile.ts] node_modules/@as-pect/assembly/assembly/index.ts --runtime minimal --debug --binaryFile output.wasm --explicitStart --use ASC_RTRACE=1 --exportTable --importMemory --transform ~/asmscript_test/node_modules/@as-pect/core/lib/transform/index.js

~/asmscript_test/node_modules/assemblyscript/lib/loader/umd/index.js:375
    const instance = new WebAssembly.Instance(module, imports);
                     ^

LinkError: WebAssembly.Instance(): Import #1 module="rtrace" function="oninit" error: function import requires a callable
    at instantiateSync (~/asmscript_test/node_modules/assemblyscript/lib/loader/umd/index.js:375:22)
    at imports (~/asmscript_test/as-pect.config.js:31:16)
    at runBinary (~/asmscript_test/node_modules/@as-pect/cli/lib/run.js:438:26)
    at ~/asmscript_test/node_modules/@as-pect/cli/lib/run.js:651:42
    at wrappedCallback (~/asmscript_test/node_modules/assemblyscript/cli/asc.js:436:14)
    at Object.main (~/asmscript_test/node_modules/assemblyscript/cli/asc.js:1058:10)
    at ~/asmscript_test/node_modules/@as-pect/cli/lib/run.js:602:17
    at Array.forEach (<anonymous>)
    at run (~/asmscript_test/node_modules/@as-pect/cli/lib/run.js:600:36)
    at Object.asp (~/asmscript_test/node_modules/@as-pect/cli/lib/index.js:64:9)
    at Object.<anonymous> (~/asmscript_test/node_modules/@as-pect/cli/lib/test.js:2:28)
    at Module._compile (node:internal/modules/cjs/loader:1108:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47
npm ERR! code 1
npm ERR! path ~/asmscript_test
npm ERR! command failed
npm ERR! command sh -c asp "--verbose"

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2021-02-10T20_34_00_932Z-debug.log
```

I'll keep trying to look at this and report back if I come up with anything useful :-)

**Comment by bcolloran** [comment]
Thanks for fixing into this bug! As-pect is now working for testing! But I did notice that in a fresh node project, the `as-pect.config.js` is written with `"--runtime": ["full"], // Acceptable values are: full, half, stub (arena), and none` rather than with the new 0.18 allowed values "incremental", "minimal", and "stub". Changing to "incremental" allows the demo tests to run, but it looks like the CLI needs at least that one tweak:

https://github.com/jtenner/as-pect/blob/67fda969e0691869bace7444543490fb3a319c13/packages/cli/init/as-pect.config.js#L17

(PS: is there a reason you didn't close this issue when you released as-pect 0.6.0? I.e., are there more things on your list to fix before announce that 0.18 is supported?)

**Comment by jtenner** [comment]
I forgot this issue exists. I'm a bit of a bird brain. Thanks for your help.
